includes/class-wc-calypso-bridge-coming-soon.php

```diff
@@ -80,19 +95,88 @@ public function should_exclude_lys_coming_soon( $exclude ) {
 	 *
 	 * @param int $old_value
 	 * @param int $new_value
-	 * @param string $option
 	 * @return void
 	 */
-	public function sync_coming_soon_option( $old_value, $new_value, $option ) {
+	public function sync_coming_soon_wpcom_to_lys( $old_value, $new_value ) {
 		if ( ! class_exists( '\Automattic\WooCommerce\Admin\Features\Features' ) || ! \Automattic\WooCommerce\Admin\Features\Features::is_enabled( 'launch-your-store' ) ) {
 			return;
 		}
 
+		$woocommerce_coming_soon = get_option( 'woocommerce_coming_soon', false );
+		$is_coming_soon_wpcom = 1 === (int) $new_value;
+
+		// Value is already set, we don't need to update option.
+		if ( ( 'yes' ===  $woocommerce_coming_soon && $is_coming_soon_wpcom ) ||
+			( 'no' === $woocommerce_coming_soon && ! $is_coming_soon_wpcom ) ) {
+			return;
+		}
+
+		// Unhook listener to prevent a loop of updating option between WPCOM <-> LYS.
+		$this->unhook_update_option_woocommerce_coming_soon();
+
 		if ( 1 ===  (int) $new_value ) {
```

Comment:
```suggestion
		if ( $is_coming_soon_wpcom ) {
```

We can reusing the `$is_coming_soon_wpcom` variable here